entities/package.py

```diff
@@ -0,0 +1,188 @@
+                 value: Value):
+        self.uuid = uuid
+        self.layer = layer
+        self.height = height
+        self.stroke_width = stroke_width
+        self.letter_spacing = letter_spacing
+        self.line_spacing = line_spacing
+        self.align = align
+        self.position = position
+        self.rotation = rotation
+        self.auto_rotate = auto_rotate
+        self.mirror = mirror
+        self.value = value
+
+    def __str__(self) -> str:
+        ret = '(stroke_text {} {}\n'.format(self.uuid, self.layer) +\
+            ' {} {} {} {}\n'.format(self.height, self.stroke_width, self.letter_spacing, self.line_spacing) +\
+            ' {} {} {}\n'.format(self.align, self.position, self.rotation) +\
+            ' {} {} {}\n)'.format(self.auto_rotate, self.mirror, self.value)
+        return ret
+
+
+class Side(EnumValue):
+    THT = 'tht'
+
+    def get_name(self) -> str:
+        return 'side'
+
+
+class Shape(EnumValue):
+    ROUND = 'round'
+    RECT = 'rect'
+
+    def get_name(self) -> str:
+        return 'shape'
+
+
+class Size():
+    def __init__(self, width: float, height: float):
+        # TODO: Are width and height good names?
+        self.width = width
+        self.height = height
+
+    def __str__(self) -> str:
+        return '(size {} {})'.format(format_float(self.width), format_float(self.height))
+
+
+class Drill(FloatValue):
+    def __init__(self, drill: float):
+        super().__init__('drill', drill)
+
+
+class FootprintPad():
+    def __init__(self, uuid: str, side: Side, shape: Shape, position: Position,
+                 rotation: Rotation, size: Size, drill: Drill):
+        self.uuid = uuid
+        self.side = side
+        self.shape = shape
+        self.position = position
+        self.rotation = rotation
+        self.size = size
+        self.drill = drill
+
+    def __str__(self) -> str:
+        ret = '(pad {} {} {}\n'.format(self.uuid, self.side, self.shape) +\
+            ' {} {} {} {}\n)'.format(self.position, self.rotation, self.size, self.drill)
+        return ret
+
+
+class Footprint():
+    def __init__(self, uuid: str, name: Name, description: Description):
+        self.uuid = uuid
+        self.name = name
+        self.description = description
+        self.pads = []  # type: List[FootprintPad]
+        self.polygons = []  # type: List[Polygon]
+        self.texts = []  # type: List[StrokeText]
+
+    def add_pad(self, pad: FootprintPad) -> None:
+        self.pads.append(pad)
+
+    def add_polygon(self, polygon: Polygon) -> None:
+        self.polygons.append(polygon)
+
+    def add_text(self, text: StrokeText) -> None:
+        self.texts.append(text)
+
+    def __str__(self) -> str:
+        ret = '(footprint {}\n'.format(self.uuid) +\
+            ' {}\n'.format(self.name) +\
+            ' {}\n'.format(self.description)
+        ret += indent_entities(self.pads)
+        ret += indent_entities(self.polygons)
+        ret += indent_entities(self.texts)
```

Comment:
Those lists of entities should probably be sorted.

Reply:
Should or are they actually sorted? And by what? UUID?

Reply:
Because in the generated output the pads are currently not sorted by UUID.

Reply:
I think by UUID, but I'm not 100% sure. Run the files through the library updater, then you can see the changes.

Reply:
I verified and they are not sorted in when using the librepcb library editor. I tried changing the names in the UI and then saving, they just get renamed but not sorted. I also tried moving them around in the file and LibrePCB leaves the order as it opened it.
Maybe it is accidentally or it is to avoid unnecessary noise in the diff.